capsules/src/process_console.rs

```diff
@@ -0,0 +1,284 @@
+//! Provides implements a text console over the UART that allows
+//! someone to control which processes are running.
+//!
+//! Protocol
+//! --------
+//!
+//! This module provides a simple text-based console to inspect and control
+//! which processes are running. The console has five commands:
+//!  - 'help' prints the available commands and arguments
+//!  - 'status' prints the current system status
+//!  - 'list' lists the current processes with their IDs and running state
+//!  - 'stop n' stops the process with ID n
+//!  - 'start n' starts the stopped process with ID n
+//!
+//! Setup
+//! -----
+//!
+//! You need a device that provides the `hil::uart::UART` trait.
+//!
+//! ```rust
+//! let console = static_init!(
+//!     ProcessConsole<usart::USART>,
+//!     ProcessConsole::new(&usart::USART0,
+//!                  115200,
+//!                  &mut console::WRITE_BUF,
+//!                  &mut console::READ_BUF,
+//!                  &mut console::COMMAND_BUF);
+//! hil::uart::UART::set_client(&usart::USART0, console);
+//! ```
+
+use core::cell::Cell;
+use core::cmp;
+use core::str;
+use kernel::capabilities::ProcessManagementCapability;
+use kernel::common::cells::TakeCell;
+use kernel::hil::uart::{self, Client, UART};
+use kernel::introspection::KernelInfo;
+use kernel::Kernel;
+use kernel::ReturnCode;
+
+/// Syscall driver number.
+pub const DRIVER_NUM: usize = 0x00000001;
```

Comment:
Should this be the same driver number as a "regular" console, or a new one?

Reply:
It's not currently a driver, so this can just be removed.